Sources/Common/Extensions/URLExtension.swift

```diff
@@ -20,6 +20,10 @@ import Foundation
 
 extension URL {
 
+    public func matches(_ other: URL) -> Bool {
```

Comment:
It would be great to cover it with unit tests, at least to manage expectations around this API. For instance:

```swift
"youtube.com".url!.matches("http://youtube.com".url!) // true
"youtube.com".url!.matches("https://youtube.com".url!) // false
```